sklearn/utils/validation.py

```diff
@@ -1057,7 +1057,7 @@ def check_X_y(
         The converted and validated y.
     """
     if y is None:
-        raise ValueError("y cannot be None")
+        raise ValueError("Expected array-like (array or non-string sequence), got None")
```

Comment:
I think including `y` in the error message is more informative. To pass the check we can do:

```python
    if y is None:
        if estimator is None:
            estimator_name = "estimator"
        else:
            estimator_name = _check_estimator_name(estimator)
        raise ValueError(
            f"{estimator_name} requires y to be passed, but the target y is None"
        )
```

Reply:
I agree, I changed the value error accordingly to your suggestion. Thanks!